app/src/main/java/org/oppia/android/app/walkthrough/topiclist/WalkthroughTopicListFragmentPresenter.kt

```diff
@@ -31,6 +33,8 @@ class WalkthroughTopicListFragmentPresenter @Inject constructor(
   private lateinit var binding: WalkthroughTopicListFragmentBinding
   private val routeToNextPage = activity as WalkthroughFragmentChangeListener
   private val orientation = Resources.getSystem().configuration.orientation
+  private val showHideProgressBarAndHeaderTextview =
+    activity as ShowHideProgressBarAndHeaderTextview
```

Comment:
Can we use Fragment here? Might need to check if this brings memory leak or not.

Reply:
@anandwana001 This is causing a crash, also like if we are implementing the interface on activity, shouldn't we use activity only? What do you say?

Reply:
Sounds correct to me.